Adopt GCController.physicalInputProfile

beidson opened the pull request:
#### 6687281bb75fb8bbebd4904a403742911ce5f033
<pre>
Adopt GCController.physicalInputProfile
<a href="https://bugs.webkit.org/show_bug.cgi?id=248110">https://bugs.webkit.org/show_bug.cgi?id=248110</a>
rdar://102365548

Reviewed by Megan Gardner.

Due to GameController framework changes in Ventura, it&apos;s possible (even likely!) for a GCController to have neither
an extendedGamepad or gamepad (which was deprecated anyways)

Instead, the widely increased number of supported devices can all be represented by physicalInputProfile.

So let&apos;s adopt that, while further tinkering with an API test that was covered by this change.

* Source/WebCore/Modules/gamepad/GamepadButton.cpp:
(WebCore::GamepadButton::GamepadButton):
* Source/WebCore/Modules/gamepad/GamepadButton.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.mm:
(WebCore::GameControllerGamepad::GameControllerGamepad):
(WebCore::GameControllerGamepad::setupElements):
(WebCore::homeButtonFromExtendedGamepad): Deleted.
(WebCore::GameControllerGamepad::setupAsExtendedGamepad): Deleted.
(WebCore::GameControllerGamepad::setupAsGamepad): Deleted.
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepadProvider.mm:
(WebCore::GameControllerGamepadProvider::prewarmGameControllerDevicesIfNecessary):
* Source/WebCore/platform/gamepad/cocoa/GameControllerSoftLink.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerSoftLink.mm:
* Tools/TestWebKitAPI/Tests/mac/HIDGamepads.mm:
* Tools/TestWebKitAPI/mac/VirtualGamepad.h:
(TestWebKitAPI::VirtualGamepad::buttonCount const):
(TestWebKitAPI::VirtualGamepad::axisCount const):

Canonical link: <a href="https://commits.webkit.org/256946@main">https://commits.webkit.org/256946@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5159312960fec34b029c7def55a2709d30475aeb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97304 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6569 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30458 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106824 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167089 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101273 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6866 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35304 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89714 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103504 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102968 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5162 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83934 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32158 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87014 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75084 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/584 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/569 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21762 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4787 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5369 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1813 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41092 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

megangardner commented on Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.h:
```diff
@@ -50,18 +48,12 @@ class GameControllerGamepad : public PlatformGamepad {
 private:
-    void setupAsExtendedGamepad();
-    void setupAsGamepad();
+    void setupElements();
```
Elements seems bit generic, is this the best name for these?

beidson replied:
It's a term of art for "individual components on a controller", and shadows the GameController API.

beidson committed:
Adopt GCController.physicalInputProfile
https://bugs.webkit.org/show_bug.cgi?id=248110
rdar://102365548

Reviewed by Megan Gardner.

Due to GameController framework changes in Ventura, it's possible (even likely!) for a GCController to have neither
an extendedGamepad or gamepad (which was deprecated anyways)

Instead, the widely increased number of supported devices can all be represented by physicalInputProfile.

So let's adopt that, while further tinkering with an API test that was covered by this change.

* Source/WebCore/Modules/gamepad/GamepadButton.cpp:
(WebCore::GamepadButton::GamepadButton):
* Source/WebCore/Modules/gamepad/GamepadButton.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.mm:
(WebCore::GameControllerGamepad::GameControllerGamepad):
(WebCore::GameControllerGamepad::setupElements):
(WebCore::homeButtonFromExtendedGamepad): Deleted.
(WebCore::GameControllerGamepad::setupAsExtendedGamepad): Deleted.
(WebCore::GameControllerGamepad::setupAsGamepad): Deleted.
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepadProvider.mm:
(WebCore::GameControllerGamepadProvider::prewarmGameControllerDevicesIfNecessary):
* Source/WebCore/platform/gamepad/cocoa/GameControllerSoftLink.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerSoftLink.mm:
* Tools/TestWebKitAPI/Tests/mac/HIDGamepads.mm:
* Tools/TestWebKitAPI/mac/VirtualGamepad.h:
(TestWebKitAPI::VirtualGamepad::buttonCount const):
(TestWebKitAPI::VirtualGamepad::axisCount const):

Canonical link: https://commits.webkit.org/256946@main
